src/compiler/diagnosticMessages.json

```diff
@@ -5013,5 +5013,41 @@
     "JSX expressions may not use the comma operator. Did you mean to write an array?": {
         "category": "Error",
         "code": 18007
+    },
+    "Private names are not allowed in variable declarations.": {
+        "category": "Error",
+        "code": 18008
+    },
+    "Private names cannot be used as parameters": {
+        "category": "Error",
+        "code": 18009
+    },
+    "Accessibility modifiers cannot be used with private names.": {
+        "category": "Error",
+        "code": 18010
+    },
+    "The operand of a delete operator cannot be a private name.": {
+        "category": "Error",
+        "code": 18011
+    },
+    "'#constructor' is a reserved word.": {
+        "category": "Error",
+        "code": 18012
+    },
+    "Property '{0}' is not accessible outside class '{1}' because it has a private name.": {
+        "category": "Error",
+        "code": 18013
+    },
+    "This usage of '{0}' refers to the private member declared in its enclosing class. While type '{1}' has a private member with the same spelling, its declaration and accessibility are distinct.": {

```

Comment:
```
The property '{0}' cannot be accessed on type '{1}' within this class because it has been shadowed by another private member with the same name.
```

Related spans:

```
The shadowing declaration of '{0}' is defined here.
```

```
The declaration of '{0}' that you probably intended to use is defined here.
```

Reply:
I personally think 'name' is fine because (1) people will know what you mean (2) it's similar to existing messages, so people will be able to skim it faster. But I don't mind 'spelling' if you feel strongly about it.

Reply:
I worry that saying 'the same name' is incorrect. The problem is that there are two distinct private names that happen to look the same.